java_grpc_library.bzl

```diff
@@ -0,0 +1,131 @@
+def _path_ignoring_repository(f):
+  if (len(f.owner.workspace_root) == 0):
+    return f.short_path
+  return f.path[len(f.owner.workspace_root)+1:]
+
+def _gensource_impl(ctx):
+  if len(ctx.attr.srcs) > 1:
+    fail("Only one src value supported", "srcs")
+  for s in ctx.attr.srcs:
+    if s.label.package != ctx.label.package:
+      print(("in srcs attribute of {0}: Proto source with label {1} should be in "
+             + "same package as consuming rule").format(ctx.label, s.label))
+  # Use .jar since .srcjar makes protoc think output will be a directory
+  srcdotjar = ctx.new_file(ctx.label.name + "_src.jar")
+
+  srcs = [f for dep in ctx.attr.srcs for f in dep.proto.direct_sources]
+  includes = [f for dep in ctx.attr.srcs for f in dep.proto.transitive_imports]
+
+  flavor = ctx.attr.flavor
+  if flavor == "normal":
+    flavor = ""
+  ctx.action(
+      inputs = [ctx.executable._java_plugin] + srcs + includes,
+      outputs = [srcdotjar],
+      executable = ctx.executable._protoc,
```

Comment:
I don't want us to make the command line ourselves, as then we have to deal with more possibilities of escaping. I'd rather us figure out what's wrong with using executable on Windows. That can happen after the merging of this PR.